generate_testdata.sh

```diff
@@ -129,9 +126,9 @@ build_kb
 # Project version 2 uses plugin go/v2 (default).
 scaffold_test_project project-v2 --project-version=2
 scaffold_test_project project-v2-multigroup --project-version=2
-scaffold_test_project project-v2-addon --project-version=2
+scaffold_test_project project-v2-addon --project-version=3 --plugins="go/v2,declarative"
```

Comment:
I do not think that we should need to inform  `go/v2` it is the layout f the project already. So, here we would inform just the declarative one.
Also, should we not keep the same name (addon)? have we any doc about it? If yes, we need a top update as well

Reply:
This again is mixing ideas from #1941 and makes no sense at all in the context of this PR.